### DISCUSSION TITLE
PSM

### DISCUSSION THREAD

**Discussion by banoz** [Zer0-bit/gaggiuino#137] [Ideas]
Hi,
have you tried to use PSM control for the pump?

**Comment by Zer0-bit** [comment]
Hi, I'm not sure what psm control means, unless it's a typo and you meant pwm?

**Reply by banoz** [reply]
It's not a typo. Pulse skip modulation.

**Reply by banoz** [reply]
https://github.com/banoz/PSM

**Reply by banoz** [reply]
You're using a RobotDyn's dimmer, it already has everything what's needed. Only the bridge has to be replaced for a diode.

**Reply by Zer0-bit** [reply]
Yeah just saw your diagram on your github, nice idea though atm the PWM works well tbh but yours might be a more portable solution since I'm adapting the current code for an stm32.
You mind joining the  project discord server where we can discuss easier?

**Reply by banoz** [reply]
Just joined.
